### ISSUE TITLE
API not very friendly to programming

### ISSUE THREAD

**Issue by renkun-ken** [yixuan/ADMM#10]
The current design of ADMM package funtions adopts the fluent style of programming, that is, the operations are done in a pipeline.

This can be friendly to interactive usage of model fitting functions, easier to read and write, rather than confused by so many parameters in one call.

However, the downside is obvious too. It is not very friendly to programming purposes. If I'm creating a general framework of modeling which supports a number of existing modeling packages (e.g., glmnet, xgboost, randomForest), it would be not easy to forward arguments to ADMM functions. 

It would be appreciated if a simple argument-based interface is provided. 

Thanks again, great work!

**Comment by yixuan** [comment]
I agree. The ADMM package was mainly developed to test the optimization algorithm, and I have basically stopped for a long time. I feel that the functionality of this package is pretty limited, so currently there is no plan to move forward. If you find it useful in some cases, feel free to fork and modify it. The C++ interface is very "classical" and I believe it is easy to provide a corresponding R wrapper for that.
